### PR TITLE
Add subtract function to calc module

### LINKED ISSUE #3: subtract関数の実装
`calc.py`に、引き算を行う`subtract`関数を実装してください。

基本的な仕様は #1 を参考にしてください。

### PR DESCRIPTION
**Pull request by t0k0sh1**
Closes #3

## Summary
- Implement subtract function in calc.py module
- Add comprehensive test cases for the subtract function
- Follow TDD approach (tests first, then implementation)

## Implementation Details
- The subtract function follows the same pattern as the add function
- Validates input types and raises ValueError for non-numeric inputs
- Handles None and boolean values explicitly

## Test plan
- [x] All tests pass (47 passed)
- [x] Lint checks pass (ruff)
- [x] Function correctly subtracts numeric values
- [x] Function raises ValueError for non-numeric inputs
- [x] Function handles edge cases (None, boolean values)

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

**Comment by t0k0sh1** [comment]
レビューありがとうございます。すべての指摘事項に対応しました。

## 対応内容

### 1. DRY原則の適用
- 重複していた入力検証ロジックを `_validate_numeric_operands` ヘルパー関数に抽出しました
- `add` 関数と `subtract` 関数の両方がこのヘルパー関数を使用するようになりました

### 2. 浮動小数点数の比較改善
- テストケースで浮動小数点数を比較する際に `pytest.approx` を使用するように修正しました
- これにより、浮動小数点数の精度問題に対してより堅牢なテストになりました

すべてのテスト（47件）が正常に通過し、Lintチェックもクリアしています。

修正はコミット b08d194 で行いました。

File: tests/python-project-tutorial/test_calc.py [hunk 1]
```diff
@@ -45,4 +45,50 @@ def test_add_正常系(self, a, b, expected):
...
+        (-1, 1, -2),
+        
+        # 限界値テスト（大きな値）
+        (2e10, 1e10, 1e10),
+        (-1e10, -2e10, 1e10),
+        (2e-10, 1e-10, 1e-10),
+    ])
+    def test_subtract_正常系(self, a, b, expected):
+        """正常系: 数値型の引数を与えた場合、正しい減算結果を返すこと"""
+        assert subtract(a, b) == expected
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply tests/python-project-tutorial/test_calc.py]: /gemini review
